2011-02-06  Tijl Coosemans <tijl@coosemans.org> 

	* config.gcc (i386-*-freebsd*): Default arch is i486.

Index: gcc/config.gcc
===================================================================
--- gcc/config.gcc	(revision 170369)
+++ gcc/config.gcc	(working copy)
@@ -2795,6 +2795,16 @@
 arch_without_sse2=no
 arch_without_64bit=no
 case ${target} in
+  i386-*-freebsd*)
+    if test $fbsd_major -ge 6; then
+      arch=i486
+    else
+      arch=i386
+    fi
+    cpu=generic
+    arch_without_sse2=yes
+    arch_without_64bit=yes
+    ;;
   i386-*-*)
     arch=i386
     cpu=i386
